.github/workflows/build_test_validate.yml

```diff
@@ -0,0 +1,30 @@
+
+name: Build and test all the Kiota packages
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+        environment:
+          name: build_test
+        strategy:
+          matrix:
+            node-version: [14.x, 16.x]
+        steps:
+        - uses: actions/checkout@v3
+        - name: Use Node.js ${{ matrix.node-version }}
+          uses: actions/setup-node@v3
+          with:
+            node-version: ${{ matrix.node-version }}
+        - run: yarn install --frozen-lockfile
```

Comment:
I'm curious to why this has been switched to yarn? is lerna pushing that requirement on us?

Reply:
yarn workspaces is optimal for dependency management and linking local dependencies.  It is popularly coupled with lerna for multi package management.

Reply:
I haven't used yarn in ages (back when it used to be much faster than npm). We should probably update the readme accordingly so people know which tools they need to get started.